### ISSUE TITLE
[iOS]: ShapeSource onPress call omitted if zoomed in or out

### ISSUE THREAD

**Issue by yossarin** [react-native-mapbox-gl/maps#336]
**Describe the bug**
ShapeSource sometimes fails to call `onPress` callback. This happens after a zoom in/zoom out of the MapView.

**To Reproduce**
The bug can be reproduced in the example app.

Modify `example/src/examples/ShapeSourceIcon.js` to:

```javascript
diff --git a/example/src/examples/ShapeSourceIcon.js b/example/src/examples/ShapeSourceIcon.js
index 8e49271..06d3b74 100755
--- a/example/src/examples/ShapeSourceIcon.js
+++ b/example/src/examples/ShapeSourceIcon.js
@@ -67,6 +67,10 @@ class ShapeSourceIcon extends React.Component {
     ...BaseExamplePropTypes,
   };
 
+  onPress = (e) => {
+    console.log("onPress callback");
+  };
+
   render() {
     return (
       <Page {...this.props}>
@@ -79,6 +83,7 @@ class ShapeSourceIcon extends React.Component {
           <MapboxGL.ShapeSource
             id="exampleShapeSource"
             shape={featureCollection}
+            onPress={this.onPress}
           >
             <MapboxGL.SymbolLayer id="exampleIconName" style={styles.icon} />
           </MapboxGL.ShapeSource>
```

Install the example app, go to `Shape Source Icon` example, and press on the icon. You should see `onPress callback` string appearing in React Native debugger.

Now, zoomout or zoomin. Try to press the icon again -> no callback. :( Sometimes, it needs a couple of zoomins/zoomouts.

**Expected behavior**

`onPress` callback should be called regardless of multiple zoomins/zoomouts on `MapView`.

**Screenshots**

**Versions (please complete the following information):**
 - Platfrom: [iOS]
 - Device: [iPhone X]
 - OS: [iOS12.4]
 - SDK Version ["@mapbox/mapbox-sdk": "^0.6.0"]
 - React Native Version [0.59.0]

**Additional context**
Might be related to #97 

I did some breakpoint debugging in XCode and had some interesting findings.

I noticed in `didTapMap` (`RCTMGLMapViewManager.m`) that a call to `visibleFeaturesInRect` function sometimes (quite often) does not return any features after some zooming in/out  (event though an Icon is clearly inside the viewport rectangle).

The culprit is this call:

```objective-c
NSArray<id<MGLFeature>> *features = [mapView visibleFeaturesInRect:hitboxRect
                                                     inStyleLayersWithIdentifiers:lala
                                                     predicate:nil];
```

So basically, after some zooming in/out `*features` array contains 0 elemets, and the icon on screen is not even considered for a "hit".

Since `mapView` function  is coming from`MGLMapView` from native SDK, I was not able to step into `.mm` files, only `.h` files.

**Comment by Gp2mv3** [comment]
We had the same issue but it seems to be solved in the last version from NPM.

We don't see the bug anymore, but I don't think something was done to correct that bug.

**Comment by kristfal** [comment]
@yossarin can you reproduce on latest master? If not, I suggest we close this and reopen if the problem surfaces again.

**Comment by yossarin** [comment]
Seemed to go away after release 7.0.4
